File: eAirFlow.Model/Models/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace eAirFlow.Model.Models
{

    public class Reservation
    {
        public int ReservationId { get; set; }

        public int? UserId { get; set; }

        public int? FlightId { get; set; }
        public int? AirportId { get; set; }
        public int? AirplaneId { get; set; }

        public DateTime? ReservationDate { get; set; }

        public int? PaymentId { get; set; }

        public string? StateMachine { get; set; }
        public int? SeatId { get; set; }
        public Seat? Seat { get; set; }

        public int? SeatClassId { get; set; }
        public SeatClass? SeatClass { get; set; }

        public int? MealTypeId { get; set; }
        public MealType? MealType { get; set; }
        public string? QrCodeBase64 { get; set; }

        public Flight? Flight { get; set; }
        public Airport? Airport { get; set; }
        public Payment? Payment { get; set; }
        public CheckIn? CheckIn { get; set; }
        public Airplane? Airplane { get; set; }




    }
}

File: UI/eairflow_mobile/lib/models/reservation.dart
import 'package:eairflow_mobile/models/airport.dart';
import 'package:eairflow_mobile/models/checkin.dart';
import 'package:eairflow_mobile/models/flight.dart';
import 'package:eairflow_mobile/models/mealtype.dart';
import 'package:eairflow_mobile/models/payment.dart';
import 'package:eairflow_mobile/models/seat.dart';

class Reservation {
  int? reservationId;
  int? userId;
  int? flightId;
  Flight? flight;

  String? reservationDate;
  int? paymentId;

  String? stateMachine;
  Seat? seat;
  String? get seatNumber => seat?.seatNumber;
  int? seatClassId;
  int? adults;
  int? children;
  MealType? mealType;
  String? luggageOption;
  String? qrCodeBase64;

  Airport? airport;
  CheckIn? checkIn;
  Payment? payment;


  Reservation();

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation()
      ..reservationId = json['reservationId']
      ..userId = json['userId']
      ..flightId = json['flightId']
      ..reservationDate = json['reservationDate']
      ..paymentId = json['paymentId']
      ..stateMachine = json['stateMachine']
      ..seat = json['seat'] != null ? Seat.fromJson(json['seat']) : null
      ..seatClassId = json['seatClassId']
      ..adults = json['adults']
      ..children = json['children']
      ..mealType = json['mealType'] != null
          ? MealType.fromJson(json['mealType'])
          : null
      ..luggageOption = json['luggageOption']
      ..qrCodeBase64 = json['qrCodeBase64']
      ..flight =
          json['flight'] != null ? Flight.fromJson(json['flight']) : null
      ..airport = json['airport'] != null ? Airport.fromJson(json['airport']) : null
      ..checkIn = json['checkIn'] != null 
     ? CheckIn.fromJson(json['checkIn']) 
     : null
     ..payment = json['payment'] != null
    ? Payment.fromJson(json['payment'])
    : null;


          
  }
}
